warehouse/query-core/src/main/java/datawave/query/function/LimitFields.java

```diff
@@ -232,57 +274,153 @@ private boolean isHit(String keyWithGrouping, Attribute<?> attr, Multimap<String
         return false;
     }
     
-    static String[] getCommonalityAndGroupingContext(String in) {
-        String[] splits = StringUtils.split(in, '.');
+    /**
+     * Return a {@link Pair} where {@link Pair#getValue0()} returns the group and {@link Pair#getValue1()} returns the instance if they can be parsed from the
+     * given key. Otherwise, an empty {@link Optional} will be returned.
+     * 
+     * @param key
+     *            the key to parse the group and instance from
+     * @return a {@link Pair} with the group and instance, or an empty {@link Optional} if they cannot be parsed
+     */
+    private Optional<Pair<String,String>> getGroupAndInstance(String key) {
```

Comment:
Might be worth considering calling it GroupAndSubgroup to align with ingest code.  No strong feelings, however

Reply:
No problem, done.